### DISCUSSION TITLE
Initial thoughts

### DISCUSSION THREAD

**Discussion by daveaglick** [kdl-org/kdl#2] [General]
I generally really like where this is heading. Figured I’d drop some initial questions/thoughts/reactions in case you find them helpful:

- I love the simple attribute syntax - I miss attributes as an extra layer of metadata in YAML and JSON.
  - Can attributes be on following lines?
  - If so, how is an attribute vs a node determined - is it the presence of a `=`?
- The use of an alpha character prefix for raw strings feels strange at first to me - I’d personally be more at home with a special character for this purpose.
- Using a `#` as the escape character also feels strange at first, especially since the comment syntax is more C/C++ - my naive expectation would be a slash escape character like most C dialects.

Looking forward to seeing where you go with this!

**Comment by zkat** [comment]
> Can attributes be on following lines?
Yup! You can do something like this:

```kdl
node-name \
  key="value" \
  another="value2"
```

> If so, how is an attribute vs a node determined - is it the presence of a `=`?

Yep!

> The use of an alpha character prefix for raw strings feels strange at first to me - I’d personally be more at home with a special character for this purpose.

>The use of an alpha character prefix for raw strings feels strange at first to me - I’d personally be more at home with a special character for this purpose.

The format is actually 100% based on Rust's string format. I figured taking an existing one and rolling with it instead of getting fancy with sigils was the best choice, and the r# thing has gone pretty well?

**Reply by daveaglick** [reply]
Ah, that makes sense about the prefixing. I'm admittedly biased towards C dialects after having worked mostly with C++, Java, and now C# so anything different feels foreign. That's obviously a narrow viewpoint so take my feedback on that one with a heaping grain of salt :)

**Reply by zkat** [reply]
the feedback is super appreciated actually, thank you! :)

I think I'll have to sleep on whether to keep `r#` or change it to a sigil or what
